txpipe/noise_maps.py

```diff
@@ -184,17 +185,31 @@ def run(self):
                 for i in range(clustering_realizations):
                     # We have computed the first half already,
                     # and we have the total map from an earlier stage
-                    half1 = ngal_split[:, b, i, 0]
-                    half2 = ngal_split[:, b, i, 1]
-
+                    #half1 = ngal_split[:, b, i, 0]
+                    #half2 = ngal_split[:, b, i, 1]
+
+                    half1 = np.zeros(npix)
+                    half2 = np.zeros_like(half1)
+
+                    if self.config['mask_in_weights']:
+                        half1 = ngal_split[:, b, i, 0]
+                        half2 =	ngal_split[:, b, i, 1]
+                    else:
+                        half1 =	(ngal_split[:, b, i, 0])/mask[pixels]
+                        half2 = (ngal_split[:, b, i, 1])/mask[pixels]
+                    
                     # Convert to overdensity.  I thought about
                     # using half the mean from the full map to reduce
                     # noise, but thought that might add covariance
                     # to the two maps, and this shouldn't be that noisy
-                    mu1 = np.average(half1, weights=mask[pixels])
-                    mu2 = np.average(half2, weights=mask[pixels])
+                    #mu1 = np.average(half1, weights=mask[pixels])
```

Comment:
I'll stop going through this now finding all the old commented out bits, but in general can you remove them?